### PR TITLE
System: Small refactorings and interrupt setup simplification

### PR DESCRIPTION
**Pull request by StarGate01**
This PR has been broken out of https://github.com/InfiniTimeOrg/InfiniTime/pull/1050.

- Unify the setup the interrupt pins in SystemTask for better clarity and readability. Allow the hardware button enable pin as well as the accelerometer interrupt pin to be configured via PinMap.h.
- Use && instead of "and" in SystemTask as per the style guide: https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/doc/coding-convention.md#coding-style=
- Add the python cache of the bootloader DFU scripts to the Git ignore list.

### CODE REVIEW

**Comment by StarGate01** [comment]
I deliberately did not fix the pre-existing formatting in `src/drivers/PinMap.h`, since this would lead to unnecessary merge conflicts with my other PRs. However, the CI task of course still complains.

**Comment by FintasticMan** [comment]
I had the same issue, it should be fixed however when #1121 is merged.

File: bootloader/ota-dfu-python/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+__pycache__
```
Comment (Riksu9000) [inline bootloader/ota-dfu-python/.gitignore:1]: This is probably an unnecessary change.

File: src/systemtask/SystemTask.cpp [hunk 2]
```diff
@@ -172,39 +172,35 @@ void SystemTask::Work() {
...
+  pinConfig.sense = NRF_GPIOTE_POLARITY_HITOLO;
+  pinConfig.pull = NRF_GPIO_PIN_PULLUP;
   nrfx_gpiote_in_init(PinMap::Cst816sIrq, &pinConfig, nrfx_gpiote_evt_handler);
+  nrfx_gpiote_in_event_enable(PinMap::Cst816sIrq, true);
+
+  // Acceleration
+  pinConfig.sense = NRF_GPIOTE_POLARITY_HITOLO;
+  pinConfig.pull = NRF_GPIO_PIN_PULLUP;
+  nrfx_gpiote_in_init(PinMap::Bma421Irq, &pinConfig, nrfx_gpiote_evt_handler);
+  nrfx_gpiote_in_event_enable(PinMap::Bma421Irq, true);
```
Comment (Riksu9000) [inline src/systemtask/SystemTask.cpp:199]: Currently this would remain unused. It's probably better to add this later when it is used.

File: bootloader/ota-dfu-python/.gitignore (hunk 1 above)
Comment (StarGate01) [reply bootloader/ota-dfu-python/.gitignore:1]: My Python installation generates these cache files when I run the DFU scripts in `bootloader/ota-dfu-python`. 

Personally I don't participial care for this change, I thought it could be useful. Should I remove it?

Comment (Riksu9000) [reply bootloader/ota-dfu-python/.gitignore:1]: I see. It does generate this folder for me too. Let's leave it then.

**Review by Riksu9000 — APPROVED** [review]

**Comment by StarGate01** [comment]
The CI format checks fail due to existing formatting issues in `PinMap.h` (see above), I am not quite sure why the LVGL simulator CI job fails - it complains of missing font object files? Could this be a transient issue?

**Comment by Riksu9000** [comment]
The simulator didn't work with the InfiniTime develop branch for a short time while the font generation feature was being added. This has also been resolved now. Rebasing will fix the issue.

**Review by JF002 — APPROVED** [review]
Looks good for me! Do you want to wait for https://github.com/InfiniTimeOrg/InfiniTime/pull/1138 before merging this one?

**Comment by StarGate01** [comment]
@JF002 thanks for the review, however I don't quite see how #1138 relates - did you mean #1121 ? In that case, yes, I can rebase once that one is merged.

**Comment by JF002** [comment]
@StarGate01 Yes, sorry, I meant https://github.com/InfiniTimeOrg/InfiniTime/pull/1121. I'll try to keep an eye on it, then!
